server/house/House.cc

```diff
@@ -1,86 +1,100 @@
-                     roomTypeStr = "Living Room";
-                } else if (room.getType() == BEDROOM){
-                     roomTypeStr = "Bedroom";
-                } else if (room.getType() == KITCHEN){
-                    roomTypeStr = "Kitchen";
-                } else {
-                    roomTypeStr = "Unknown";
-                }
-                outFile << "Room Type: " << roomTypeStr << std::endl;
-
-                const auto& sensors = room.getSensors();
-                for (const auto& sensor : sensors) {
-                    outFile << "Sensor Type: " << getSensorTypeName(*sensor) << ", ";
-                    outFile << "Value: " << sensor->getValue() << std::endl;
-                }
-                outFile << std::endl;
-            }
-        } catch (const std::ios_base::failure& e) {
-            std::cerr << "I/O error while writing data: " << e.what() << std::endl;
-            outFile.close();
-            return;
-        }
-
-        outFile.close();
-    }
-
-    std::string House::getSensorTypeName(const Sensor &sensor) {
-        const std::type_info& ti = typeid(sensor);
-        if (ti == typeid(TemperatureSensor)) {
-            return "TemperatureSensor";
-        } else if (ti == typeid(HumiditySensor)) {
-            return "HumiditySensor";
-        } else if (ti == typeid(MonoxideSensor)) {
-            return "MonoxideSensor";
-        }
-        return "UnknownSensor";
-        }
-
-}
-
+#include "House.h"
+#include <fstream>
+#include <iostream>
+
+namespace smart_home {
+    House::House(int numRooms) {
+        if (numRooms > MAX_ROOMS) {
+            std::cerr << "Error: Number of rooms exceeds maximum number of rooms.\n";
+            return;
+        }
+        
+        initializeRooms();
+    }
+
+    void House::initializeRooms() {
+        for (size_t i = 0; i < rooms.size(); ++i) {
+            auto type = static_cast<RoomType>(i % 3);
+            rooms[i] = Room(type);
+        }
+        for (auto& room : rooms) {
+            room.addTemperatureSensor("TempSensor_" + std::to_string(room.getType()));
+            room.addHumiditySensor("HumiditySensor_" + std::to_string(room.getType()));
+            if (room.getType() == RoomType::KITCHEN) {
+                room.addMonoxideSensor("MonoxideSensor_" + std::to_string(room.getType()));
+            }
+            room.generateRandomSensorValues();
+        }
+    }
+    
+    std::array<Room, 3> House::getRooms() const {
+        return rooms;
+    }
+
+    Room House::getRoom(RoomType roomType) const {
+        for (const auto& room : rooms) {
+            if (room.getType() == roomType) {
+                return room;
+            }
+        }
+        std::cerr << "Room not found." << std::endl;
+        return {};
+    }
+
+    void House::regenerateSensorValue(){
+        for (const auto& room : rooms){
+            room.generateRandomSensorValues();
+        }
+    }
+
+    void House::writeDataToFile() {
```

Comment:
No need to change this now, just to keep in mind. As we discussed in SOLID presentation, it should not be a "house"s responsibility to write to file, you should have a dedicate class that does the write/read.